internal/cli/atlas/deployments/setup.go

```diff
@@ -508,12 +553,33 @@ func (opts *SetupOpts) validateDeploymentTypeFlag() error {
 		return errFlagTypeRequired
 	}
 
+	if !opts.IsLocalDeploymentType() && opts.bindIPAll {
+		return errIncompatibleDeploymentType
+	}
+
 	if opts.DeploymentType != "" && !strings.EqualFold(opts.DeploymentType, options.AtlasCluster) && !strings.EqualFold(opts.DeploymentType, options.LocalCluster) {
 		return fmt.Errorf("%w: %s", errInvalidDeploymentType, opts.DeploymentType)
 	}
 
 	return nil
 }
+
+func (opts *SetupOpts) validateBindIPAllFlag() error {
```

Comment:
consider that cobra now supports flags that are required together and flags that are mutually exclusive even for grouped flags https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#flag-groups